### PR TITLE
Adding SQL AutoDR support

### PR DESCRIPTION
**Pull request by franktsa**
Adding support for AutoDr functionality, which is a configuration object under the server. Operations include:
•Create (PUT)
•Read (GET)
•Update / Failover (POST)
•Delete (DELETE)

CR: http://codeflow/dashboard/review/display/franktsa-c410159bf86f4ca8b7a9f2c6b5768b9f

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@franktsa__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (franktsa). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/ServerDisasterRecoveryConfigurationTests.ps1 [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+#>
+function Test-ServerDisasterRecoveryConfigurationInternal ($location1 = "North Europe", $location2 = "Southeast Asia")
+{
+	# Setup
+	$rg1 = Create-ResourceGroupForTest $location1
+	$rg2 = Create-ResourceGroupForTest $location2
+	
+	try
+	{
+		$server1 = Create-ServerForTest $rg1 "12.0" $location1
```
Comment (hovsepm) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/ServerDisasterRecoveryConfigurationTests.ps1]: @franktsa please align indentation. Replace tab with 4 spaces in the file.

File: src/ResourceManager/Sql/Commands.Sql.Test/packages.config [hunk 2]
```diff
@@ -2,13 +2,14 @@
 <packages>
   <package id="Hyak.Common" version="1.0.3" targetFramework="net45" />
   <package id="Microsoft.Azure.Common" version="2.1.0" targetFramework="net45" />
+  <package id="Microsoft.Azure.Common.Authentication" version="1.4.1-preview" targetFramework="net45" />
```
Comment (hovsepm) [inline src/ResourceManager/Sql/Commands.Sql.Test/packages.config]: Common authentication was moved to powershell repo. Please reference it from there

File: src/ResourceManager/Sql/Commands.Sql/packages.config [hunk 3]
```diff
@@ -2,10 +2,11 @@
 <packages>
   <package id="Hyak.Common" version="1.0.3" targetFramework="net45" />
   <package id="Microsoft.Azure.Common" version="2.1.0" targetFramework="net45" />
+  <package id="Microsoft.Azure.Common.Authentication" version="1.4.1-preview" targetFramework="net45" />
```
Comment (hovsepm) [inline src/ResourceManager/Sql/Commands.Sql/packages.config]: same comment. use it form powershell repo

**Comment by hovsepm** [comment]
on-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/543/

File: src/ResourceManager/Sql/Commands.Sql/Commands.Sql.csproj [hunk 4]
```diff
@@ -269,6 +280,10 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.Azure.Common.2.1.0\lib\net45\Microsoft.Azure.Common.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Azure.Common.Authentication">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Common.Authentication.1.4.1-preview\lib\net45\Microsoft.Azure.Common.Authentication.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Sql/Commands.Sql/Commands.Sql.csproj]: remove this. You need to reference solution in the repo

File: src/ResourceManager/Sql/Commands.Sql/Commands.Sql.csproj [hunk 5]
```diff
@@ -322,6 +337,10 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.Bcl.Async.1.0.168\lib\net40\Microsoft.Threading.Tasks.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.WindowsAzure.Configuration, Version=1.8.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.WindowsAzure.ConfigurationManager.1.8.0.0\lib\net35-full\Microsoft.WindowsAzure.Configuration.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Sql/Commands.Sql/Commands.Sql.csproj]: use configuration manager version 3.2.0

**Comment by hovsepm** [comment]
new on-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/543/

**Comment by markcowl** [comment]
@hovsepm @franktsa The common.authentication reference still needs to be removed

**Comment by hovsepm** [comment]
last on-demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/546/

**Comment by markcowl** [comment]
@franktsa What are the plans for adding help for these cmdlets?  Also, can you please send a PR with updates to the ChangeLog for these cmdlets?

**Pull request merged**